SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in

```diff
@@ -17,6 +17,16 @@ set(SOFA_HAVE_GLEW "@SOFA_HAVE_GLEW@")
 set(SOFA_HAVE_BOOST "@SOFA_HAVE_BOOST@")
 set(SOFA_NO_OPENGL "@SOFA_NO_OPENGL@")
 set(SOFA_HAVE_GTEST "@SOFA_HAVE_GTEST@")
+set(SOFA_BUILD_TESTS "@SOFA_BUILD_TESTS@")
+
+set(SOFA_BUILD_METIS "@SOFA_BUILD_METIS@")
+set(PLUGIN_SOFAPYTHON "@PLUGIN_SOFAPYTHON@")
+set(SOFA_USE_MASK "@SOFA_USE_MASK@")
+set(SOFA_WITH_DEVTOOLS "@SOFA_WITH_DEVTOOLS@")
+set(SOFA_WITH_THREADING "@SOFA_WITH_THREADING@")
+set(SOFA_WITH_DEPRECATED_COMPONENTS "@SOFA_WITH_DEPRECATED_COMPONENTS@")
+set(SOFA_BUILD_TUTORIALS "@SOFA_BUILD_TUTORIALS@")
+set(MODULE_SOFASPARSESOLVER "@MODULE_SOFASPARSESOLVER@")
```

Comment:
I'm not sure how precise it is needed but what about
```cmake
find_package(SofaSparseSolver)
if(SofaSparseSolver_FOUND)
      if(SofaSparseSolver_OTHEROPTIONS)
                 ....
     endif()
endif()
```

Reply:
Very good point... and obvious answer.
I was looking at how SoftRobots and STLIB looks for its dependencies, and didn't think of replacing the search for the CMake option "MODULE_SOFAXXX" and "PLUGIN_XXX" with the "XXX_FOUND" generated in the find_package...

For all the other cases (SOFA_WITH_*, SOFA_BUILD_*, SOFA_USE_*, ...), are you OK with exporting them in the SofaFrameworkConfig.cmake?

Reply:
While you're right that it's how the module import should be done, a find_package(SofaSparseSolver) in an out-of-tree plugin build results in the following error:

```
CMake Error at CMakeLists.txt:11 (find_package):
  Found package configuration file:

    /home/bruno/dev/SOFA/sofa/build/install/lib/cmake/SofaSparseSolver/SofaSparseSolverConfig.cmake

  but it set SofaSparseSolver_FOUND to FALSE so package "SofaSparseSolver" is
  considered to be NOT FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing:
  SofaBaseLinearSolver SofaGeneralLinearSolver SofaImplicitOdeSolver
  SofaSimpleFem metis csparse
```

It looks like the module's dependency targets are not properly referenced. I'll investigate, either in this PR or in another one..(?)
Other than that I removed the PLUGIN_SOFAPYTHON variable export & to me this PR is ready for review